app/controllers/api/shops_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Api::ShopsController < Api::BaseController
+  before_action :doorkeeper_authorize!, only: %i[update]
+  def update
+    begin
+      params.require(:id)
+      params.require(:name)
+      params.require(:address)
+      params.require(:description)
```

Comment:
The `Api::ShopsController` is using a manual way to require parameters and validate them. Consider using ActiveModel Serializers or similar mechanisms for parameter validation to make the code cleaner and more maintainable.